app/controllers/HomeController.scala

```diff
@@ -5,16 +5,17 @@ import javax.inject._
 import com.typesafe.scalalogging.LazyLogging
 import play.api.mvc._
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
```

Comment:
Any one more thing is, if you are upgrading the Play 2.4/2.5 to 2.6, you will receive a lot of compile warning saying that:

```
object Execution in package concurrent is deprecated (since 2.6.0): Please see https://www.playframework.com/documentation/2.6.x/Migration26#Execution
```

In your demo:

```
    logger.info("Logging")
    Future {
      logger.info("Logging in the Future!")
    }
```

I think `Future {}` created a new thread, and if you handle this by the Scala default dispatcher, the MDC will not be updated. The MDC will be only updated if we handle the thread by our customized Akka dispatcher named `MDCPropagatingDispatcher `.


Right now I am fixing these warning to embracing the Dependency injection.